hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/BaseFileOnlyRelation.scala

```diff
@@ -60,6 +60,12 @@ class BaseFileOnlyRelation(sqlContext: SQLContext,
                                     tableSchema: HoodieTableSchema,
                                     requiredSchema: HoodieTableSchema,
                                     filters: Array[Filter]): HoodieUnsafeRDD = {
+    if (!internalSchema.isEmptySchema) {
+      // it is safe to enable vectorizedReader
+      sqlContext.sparkSession.sessionState.conf.setConfString("spark.sql.parquet.recordLevelFilter.enabled", "false")
```

Comment:
@xiarixiaoyao : Can you add the comment in code why recordLevelFilter.enabled is turned off in this case ?

Reply:
we no  need  recordLevelFilter.enabled  for  snapshot read,    this filter only for incremental query，  mor incremental query need to filter data through file level filters
i also  post another pr to  fixed  regression problem https://github.com/apache/hudi/pull/5168   once  merged  will remove  those code.